python_dwd/additionals/helpers.py

```diff
@@ -111,109 +116,101 @@ def metaindex_for_1minute_data(parameter: Parameter,
+    return file
 
-        geo_file.columns = [GERMAN_TO_ENGLISH_COLUMNS_MAPPING.get(
-            name.strip().upper(), name.strip().upper())
-            for name in geo_file.columns]
 
-        par_file.columns = [GERMAN_TO_ENGLISH_COLUMNS_MAPPING.get(
-            name.strip().upper(), name.strip().upper())
-            for name in par_file.columns]
+def combine_geo_and_par_file_to_metadata_df(metadata_file):
+    with zipfile.ZipFile(metadata_file) as zip_file:
+        files_in_zipfile = zip_file.namelist()
 
-        geo_file = geo_file.iloc[[-1], :]
-        par_file = par_file.loc[:, [FROM_DATE_NAME, TO_DATE_NAME]].dropna()
+        files_of_geo_and_par = {}
+        dfs_of_geo_and_par = {}
 
-        geo_file[FROM_DATE_NAME] = par_file[FROM_DATE_NAME].min()
-        geo_file[TO_DATE_NAME] = par_file[TO_DATE_NAME].max()
+        for file in files_in_zipfile:
+            if find_all_matchstrings_in_string(file.lower(), METADATA_1MIN_GEO_MATCHSTRINGS):
+                files_of_geo_and_par["geo_file"] = file
+            elif find_all_matchstrings_in_string(file.lower(), METADATA_1MIN_PAR_MATCHSTRINGS):
+                files_of_geo_and_par["par_file"] = file
 
-        geo_file = geo_file.loc[:, METADATA_1MIN_COLUMNS]
+        for filetype, file in files_of_geo_and_par.items():
+            with zip_file.open(file) as file_opened:
+                try:
+                    df = parse_zipped_data_into_df(file_opened)
+                except UnicodeDecodeError:
+                    df = parse_zipped_data_into_df(file_opened,
+                                                   engine='python')
 
-        metadata_df = metadata_df.append(geo_file,
-                                         ignore_index=True)
+            df.columns = [GERMAN_TO_ENGLISH_COLUMNS_MAPPING.get(
+                name.strip().upper(), name.strip().upper())
+                for name in df.columns]
 
-    columns = metadata_df.columns
-    META_INDEX_DTYPES = {columns[0]: int,
-                         columns[1]: datetime64,
-                         columns[2]: datetime64,
-                         columns[3]: float,
-                         columns[4]: float,
-                         columns[5]: float,
-                         columns[6]: str}
-    metadata_df = metadata_df.astype(META_INDEX_DTYPES)
+            dfs_of_geo_and_par[filetype] = df
 
-    metadata_df = metadata_df.sort_values(
-        STATION_ID_NAME).reset_index(drop=True)
+    dfs_of_geo_and_par["geo_file"] = dfs_of_geo_and_par["geo_file"].iloc[[-1], :]
 
-    return metadata_df
+    dfs_of_geo_and_par["par_file"] = dfs_of_geo_and_par["par_file"].loc[:, [FROM_DATE_NAME, TO_DATE_NAME]].dropna()
 
+    dfs_of_geo_and_par["geo_file"][FROM_DATE_NAME] = dfs_of_geo_and_par["par_file"][FROM_DATE_NAME].min()
+    dfs_of_geo_and_par["geo_file"][TO_DATE_NAME] = dfs_of_geo_and_par["par_file"][TO_DATE_NAME].max()
 
-def parse_zipped_data_into_df(file_opened, engine: str = 'c'):
```

Comment:
missing type hints and add a small docstring

Reply:
Did.